src/Arcus.Observability.Telemetry.AspNetCore/Extensions/ILoggerExtensions.cs

```diff
@@ -44,6 +45,65 @@ public static void LogRequest(this ILogger logger, HttpRequest request, HttpResp
+        /// <param name="request">The incoming HTTP request that was processed.</param>
+        /// <param name="response">The outgoing HTTP response that was created.</param>
+        /// <param name="measurement">The instance to measure the latency duration of the HTTP request.</param>
+        /// <param name="context">The context that provides more insights on the tracked HTTP request.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="logger"/>, <paramref name="request"/>, or <paramref name="measurement"/> is <c>null</c></exception>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when the <paramref name="response"/>'s status code is outside the 0-999 range inclusively.</exception>
+        public static void LogRequest(
+            this ILogger logger,
+            HttpRequest request,
+            HttpResponse response,
+            DurationMeasurement measurement,
+            Dictionary<string, object> context = null)
+        {
+            Guard.NotNull(logger, nameof(logger), "Requires a logger instance to track telemetry");
+            Guard.NotNull(request, nameof(request), "Requires a HTTP request instance to track a HTTP request");
+            Guard.NotNull(response, nameof(response), "Requires a HTTP response instance to track a HTTP request operation");
+            Guard.NotNull(measurement, nameof(measurement), "Requires an measurement instance to time the duration of the HTTP request");
+
+            context = context ?? new Dictionary<string, object>();
+
+            LogRequest(logger, request, response, measurement.Elapsed, measurement.StartTime, context);
+        }
+
+        /// <summary>
+        /// Logs an HTTP request.
+        /// </summary>
+        /// <param name="logger">The logger to track the telemetry.</param>
+        /// <param name="request">The incoming HTTP request that was processed.</param>
+        /// <param name="response">The outgoing HTTP response that was created.</param>
+        /// <param name="duration">The duration of the HTTP request processing operation.</param>
+        /// <param name="startTime">The time when the HTTP request was received.</param>
+        /// <param name="context">The context that provides more insights on the tracked HTTP request.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="logger"/> or <paramref name="request"/> is <c>null</c></exception>
+        /// <exception cref="ArgumentOutOfRangeException">
+        ///     Thrown when the <paramref name="response"/>'s status code is outside the 0-999 range inclusively,
+        ///     or the <paramref name="duration"/> is a negative time range.
+        /// </exception>
+        public static void LogRequest(
+            this ILogger logger,
+            HttpRequest request,
+            HttpResponse response,
+            TimeSpan duration,
+            DateTimeOffset startTime,
```

Comment:
Nitpicking:  it might be more logical to specify the startTime parameter before the duration parameter ?

Reply:
Ah, yes, I based myself on how we track Azure Service Bus requests, but apparently, we do indeed place the start time in dependency tracking first.
Hmm, shame that we have missed it in Azure Service Bus request tracking 😢 . However, we recommend using the `DurationMeasurement` either way, so I guess it's not so bad.

Ok, let's change this! 👍